### PR TITLE
Polish Cache Unit Tests

### PR DESCRIPTION
**Pull request by jasoncoolmax**
- added more UTs to increase test coverage
- divide big UT functions into small UT functions
- added utility functions to improve test code quality

### CODE REVIEW

File: MSAL/test/unit/MSALTokenCacheAccessorTests.m [hunk 1]
```diff
@@ -121,284 +136,260 @@ - (void)tearDown {
...
-    [_cache saveAccessAndRefreshToken:requestParam response:_testTokenResponse context:nil error:nil];
+    //there should be two ATs and RTs in cache
+    NSArray <MSALAccessTokenCacheItem *> *atsInCache = [_cache.dataSource getAccessTokenItemsWithKey:nil context:nil error:nil];
+    NSArray <MSALRefreshTokenCacheItem *> *rtsInCache = [_cache.dataSource allRefreshTokens:nil context:nil error:nil];
+    XCTAssertEqual(atsInCache.count, 2);
+    XCTAssertEqual(rtsInCache.count, 2);
     
-    //retrieve RT
-    MSALRefreshTokenCacheItem *rtItemInCache = [_cache findRefreshToken:requestParam context:nil error:nil];
+    atsInCache = [atsInCache sortedArrayUsingComparator:^NSComparisonResult(id a, id b) {
```
Comment (RPangrle) [inline MSAL/test/unit/MSALTokenCacheAccessorTests.m]: You don't have to do this, but it's good to know. If you implement both `-isEqual:` and `-hash` on a class you can dump them into NSSet and is isEqual on the sets to see if the two are the same. :)

File: MSAL/test/unit/MSALTokenCacheAccessorTests.m [hunk 2]
```diff
@@ -121,284 +136,260 @@ - (void)tearDown {
...
     requestParam.user = _user1;
-    
-    MSALRequestParameters *requestParam2 = [MSALRequestParameters new];
-    requestParam2.unvalidatedAuthority = _testAuthority;
-    requestParam2.clientId = _testClientId;
-    [requestParam2 setScopesFromArray:@[@"User.Read"]];
-    requestParam2.user = _user2;
-    
+    [requestParam setScopesFromArray:@[@"User.Read", @"scope.notexist"]];
+    NSError *error;
```
Comment (RPangrle) [inline MSAL/test/unit/MSALTokenCacheAccessorTests.m]: initialize this to nil

File: MSAL/test/unit/MSALTokenCacheAccessorTests.m [hunk 3]
```diff
@@ -121,284 +136,260 @@ - (void)tearDown {
...
-    //there should be two RTs in cache
-    XCTAssertEqual([_cache.dataSource allRefreshTokens:nil context:nil error:nil].count, 2);
-    
-    //retrieve AT
-    MSALAccessTokenCacheItem *atItemInCache = [_cache findAccessToken:requestParam context:nil authorityFound:nil error:nil];
-    
-    //compare AT with the AT retrieved from cache
-    XCTAssertEqualObjects([atItem tokenCacheKey:nil].service, [atItemInCache tokenCacheKey:nil].service);
-    XCTAssertEqualObjects([atItem tokenCacheKey:nil].account, [atItemInCache tokenCacheKey:nil].account);
+    NSError *error;
```
Comment (RPangrle) [inline MSAL/test/unit/MSALTokenCacheAccessorTests.m]: Initialize this to nil.

File: MSAL/test/unit/MSALTokenCacheAccessorTests.m (hunk 1 above)
Comment (jasoncoolmax) [inline MSAL/test/unit/MSALTokenCacheAccessorTests.m]: Nice! You mentioned that before, let me put it into practice :)

File: MSAL/test/unit/utils/MSALTestTokenCacheItemUtil.m [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    RETURN_NO_ON_OBJECT_UNEQUAL(self.user.displayableId, item.user.displayableId);
+    RETURN_NO_ON_OBJECT_UNEQUAL(self.user.name, item.user.name);
+    RETURN_NO_ON_OBJECT_UNEQUAL(self.user.identityProvider, item.user.identityProvider);
+    RETURN_NO_ON_OBJECT_UNEQUAL(self.user.uid, item.user.uid);
+    RETURN_NO_ON_OBJECT_UNEQUAL(self.user.utid, item.user.utid);
+    RETURN_NO_ON_OBJECT_UNEQUAL(self.user.environment, item.user.environment);
+    RETURN_NO_ON_OBJECT_UNEQUAL(self.user.userIdentifier, item.user.userIdentifier);
+    RETURN_NO_ON_OBJECT_UNEQUAL(self.tenantId, item.tenantId);
+    RETURN_NO_ON_OBJECT_UNEQUAL([NSNumber numberWithBool:self.isExpired], [NSNumber numberWithBool:item.isExpired]);
+    RETURN_NO_ON_OBJECT_UNEQUAL([self tokenCacheKey:nil].service, [item tokenCacheKey:nil].service);
```
Comment (RPangrle) [inline MSAL/test/unit/utils/MSALTestTokenCacheItemUtil.m]: Remove the tokenCacheKey checks, as those aren't properties on the object.

**Pull request merged**